// ng2-http-loader/loader/ng2-http-loader.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var ng2_http_loader_component_1 = require('./ng2-http-loader.component');
var ng2_http_1 = require('./ng2-http');
var http_1 = require('@angular/http');
var ng2_http_loader_emitter_1 = require('./ng2-http-loader-emitter');
var Ng2HttpLoaderModule = (function () {
    function Ng2HttpLoaderModule() {
    }
    Ng2HttpLoaderModule = __decorate([
        core_1.NgModule({
            imports: [],
            exports: [ng2_http_loader_component_1.Ng2HttpLoaderComponent],
            declarations: [ng2_http_loader_component_1.Ng2HttpLoaderComponent],
            providers: [
                ng2_http_loader_emitter_1.Ng2HttpLoaderEmitter,
                { provide: http_1.Http,
                    useFactory: ng2_http_1.httpLoaderFactory,
                    deps: [http_1.XHRBackend, http_1.RequestOptions, core_1.Injector] }
            ]
        }), 
        __metadata('design:paramtypes', [])
    ], Ng2HttpLoaderModule);
    return Ng2HttpLoaderModule;
}());
exports.Ng2HttpLoaderModule = Ng2HttpLoaderModule;
