// ng2-component-spinner/dist/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@angular/core");
const common_1 = require("@angular/common");
const spinner_component_1 = require("./src/spinner.component");
__export(require("./src/spinner.component"));
let SpinnerComponentModule = SpinnerComponentModule_1 = class SpinnerComponentModule {
    static forRoot() {
        return {
            ngModule: SpinnerComponentModule_1,
        };
    }
};
SpinnerComponentModule = SpinnerComponentModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            common_1.CommonModule
        ],
        declarations: [
            spinner_component_1.SpinnerComponent
        ],
        exports: [
            spinner_component_1.SpinnerComponent
        ]
    })
], SpinnerComponentModule);
exports.SpinnerComponentModule = SpinnerComponentModule;
var SpinnerComponentModule_1;
